console/src/components/Services/RemoteSchema/Permissions/RSPInput.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from 'react';
+import { GraphQLInputField } from 'graphql';
+import Pen from './Pen';
+import { useDebouncedEffect } from '../../../../hooks/useDebounceEffect';
+import { isNumberString } from '../../../Common/utils/jsUtils';
+
+interface RSPInputProps {
+  k: string;
+  editMode: boolean;
+  value: string | Record<string, any>;
+  v: GraphQLInputField;
+  setArgVal: (v: Record<string, unknown>) => void;
+  setEditMode: (b: boolean) => void;
+}
+const RSPInput: React.FC<RSPInputProps> = ({
+  k,
+  editMode,
+  value,
+  setArgVal,
+  v,
+  setEditMode,
+}) => {
+  const [localValue, setLocalValue] = useState<string>(value as string);
+
+  // TODO check more on the browser console error message
+  useDebouncedEffect(
+    () => {
+      if (
+        (v?.type?.inspect() === 'Int' || v?.type?.inspect() === 'Int!') &&
+        localValue &&
+        isNumberString(localValue)
+      ) {
+        if (localValue === '0') return setArgVal({ [v?.name]: 0 });
+        return setArgVal({ [v?.name]: Number(localValue) });
+      }
+
+      setArgVal({ [v?.name]: localValue });
+    },
+    500,
+    [localValue]
+  ); // ignore continues onChange events till the user finish typing
+  // TODO support different input types :?
+  return (
+    <>
+      <label htmlFor={k}> {k}:</label>
+      {editMode ? (
+        <>
+          <input
+            value={localValue}
+            style={{
+              border: 0,
+              borderBottom: '2px dotted black',
+              borderRadius: 0,
+            }}
+            onChange={e => setLocalValue(e.target.value)}
+          />
+        </>
+      ) : (
+        <button onClick={() => setEditMode(true)}>
+          <Pen />
+        </button>
+      )}
+    </>
+  );
+};
+
+export default React.memo(RSPInput);
```

Comment:
Change to the named default export (`const X = React.memo(Y); export default X`), so that auto import works, and we use it as `X` in other places.